org/postgresql/jdbc2/AbstractJdbc2Statement.java

```diff
@@ -1866,15 +1847,191 @@ else if (in instanceof InputStream)
             case Types.DISTINCT:
                 bindString(parameterIndex, in.toString(), Oid.UNSPECIFIED);
                 break;
-	        case Types.OTHER:
-	            if (in instanceof PGobject)
-	                setPGobject(parameterIndex, (PGobject)in);
-	            else
+            case Types.OTHER:
+                if (in instanceof PGobject)
+                    setPGobject(parameterIndex, (PGobject)in);
+                else
                     bindString(parameterIndex, in.toString(), Oid.UNSPECIFIED);
-	            break;
-	        default:
-	            throw new PSQLException(GT.tr("Unsupported Types value: {0}", targetSqlType), PSQLState.INVALID_PARAMETER_TYPE);
+                break;
+            default:
+                throw new PSQLException(GT.tr("Unsupported Types value: {0}", targetSqlType), PSQLState.INVALID_PARAMETER_TYPE);
+        }
+    }
+
+    private static String asString(final Clob in) throws SQLException {
+        return in.getSubString(1, (int) in.length());
+    }
+
+    private static int castToInt(final Object in) throws SQLException {
+        try {
+            if (in instanceof String)
+                return Integer.parseInt((String) in);
+            if (in instanceof Number)
+                return ((Number) in).intValue();
+            if (in instanceof java.util.Date)
+                return (int) ((java.util.Date) in).getTime();
```

Comment:
Looking at the spec, no we don't have to do this, but as this was the current behaviour changing it may break existing code.